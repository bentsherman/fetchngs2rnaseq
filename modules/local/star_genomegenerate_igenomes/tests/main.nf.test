nextflow_process {

    name "Test Process STAR_GENOMEGENERATE_IGENOMES"
    script "../main.nf"
    process "STAR_GENOMEGENERATE_IGENOMES"

    test("fasta with gtf") {

        when {
            process {
                """
                input[0] = Channel.of([file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)])
                input[1] = Channel.of([file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true)])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(file(process.out.index[0][1]).listFiles().collect { it.getName() }.sort().toString()).match("fasta_gtf_index") },
                { assert snapshot(process.out.versions).match("fasta_gtf_versions") }
            )
        }
    }

    test("fasta with gtf - stub") {

        options '-stub'

        when {
            process {
                """
                input[0] = Channel.of([file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)])
                input[1] = Channel.of([file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true)])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(file(process.out.index[0][1]).listFiles().collect { it.getName() }.sort().toString()).match("fasta_gtf_stub_index") },
                { assert snapshot(process.out.versions).match("fasta_gtf_stub_versions") }
            )
        }
    }

    test("fasta no gtf") {

        when {
            process {
                """
                input[0] = Channel.of([file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)])
                input[1] = Channel.of([ ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(file(process.out.index[0][1]).listFiles().collect { it.getName() }.sort().toString()).match("fasta_index") },
                { assert snapshot(process.out.versions).match("fasta_versions") }
            )
        }

    }

    test("fasta no gtf - stub") {

        options '-stub'

        when {
            process {
                """
                input[0] = Channel.of([file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)])
                input[1] = Channel.of([ ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(file(process.out.index[0][1]).listFiles().collect { it.getName() }.sort().toString()).match("fasta_stub_index") },
                { assert snapshot(process.out.versions).match("fasta_stub_versions") }
            )
        }
    }
}
