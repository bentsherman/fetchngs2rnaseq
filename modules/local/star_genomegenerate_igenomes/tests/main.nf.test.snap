{
    "fasta_gtf_versions": {
        "content": [
            [
                "versions.yml:md5,562a70f12dd9e695449b59fbf604e98d"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-05T10:28:05.069709"
    },
    "fasta_stub_versions": {
        "content": [
            [
                "versions.yml:md5,562a70f12dd9e695449b59fbf604e98d"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-05T10:31:12.153651"
    },
    "fasta_gtf_stub_index": {
        "content": [
            "[]"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-05T10:28:16.309869"
    },
    "fasta_gtf_stub_versions": {
        "content": [
            [
                "versions.yml:md5,562a70f12dd9e695449b59fbf604e98d"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-05T10:28:16.337298"
    },
    "fasta_index": {
        "content": [
            "[]"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-05T10:30:56.504744"
    },
    "fasta_versions": {
        "content": [
            [
                "versions.yml:md5,562a70f12dd9e695449b59fbf604e98d"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-05T10:30:56.575367"
    },
    "fasta_gtf_index": {
        "content": [
            "[]"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-05T10:28:05.031195"
    },
    "fasta_stub_index": {
        "content": [
            "[]"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-05T10:31:12.12097"
    }
}